src/bios/f-macros/src/lib.rs

```diff
@@ -0,0 +1,227 @@
+#![feature(proc_macro_span)]
+#![feature(proc_macro_quote)]
+extern crate proc_macro2;
+
+use std::fs;
+use std::path;
+use std::path::PathBuf;
+
+use proc_macro2::{Span, TokenStream};
+use quote::quote;
+use syn::{Ident, Item, ItemFn, parse_file, parse_macro_input, Stmt};
+use syn::__private::str;
+use syn::spanned::Spanned;
+
+/// This procedural macro will generate a function that will build the IDT from
+/// the module where all interrupts are defined.
+/// This function will then have to be called from the main function.
+/// ```
+/// generate_idt();
+/// ```
```

Comment:
```suggestion
/// This function will then have to be called from the main function.
///
/// ```
/// generate_idt();
/// ```
```